backend/src/routes/api/k8s/pass-through.ts

```diff
@@ -0,0 +1,95 @@
+import https, { RequestOptions } from 'https';
+import { K8sResourceCommon, K8sStatus, KubeFastifyInstance } from '../../../types';
+
+type PassThroughData = {
+  method: string;
+  requestData: string;
+  url: string;
+};
+
+const isK8sStatus = (data: Record<string, unknown>): data is K8sStatus => data.kind === 'Status';
+
+export const passThrough = async (
+  data: PassThroughData,
+  fastify: KubeFastifyInstance,
+): Promise<{ response: K8sResourceCommon }> => {
+  const kc = fastify.kube.config;
+  const { method, requestData, url } = data;
+
+  // TODO: Remove when bug is fixed - https://issues.redhat.com/browse/HAC-1825
+  let safeURL = url;
+  if (method.toLowerCase() === 'post') {
+    // Core SDK builds the wrong path for k8s -- can't post to a resource name; remove the name from the url
+    // eg: POST /.../configmaps/my-config-map => POST /.../configmaps
+    const urlParts = url.split('/');
+    urlParts.pop();
+    safeURL = urlParts.join('/');
+  }
+
+  return new Promise((resolve, reject) => {
+    const kubeOptions: Parameters<typeof kc.applyToRequest>[0] = { url: safeURL };
+    kc.applyToRequest(kubeOptions).then(() => {
```

Comment:
This needs to be a bearer token from the user (from the oauth proxy). This is going to be the same service account that started the backend.

Reply:
Talking with @lucferbux -- we need to write a fastify plugin to be `dev.ts` (like `kube.ts`) that will just manage pulling the current logged in user (from `kube.ts`) and replace the `x-forwarded-...` header (it likely won't exist, but if it does toss it out and replace it). 

That way when we make local development calls the same way we would do on-cluster as the service account used to do. `x-forwarded-...` header will have the token OF THE USER, not the service account when this PR merges (in both local & on cluster -- so it is the same experience).